datastore/query.js

```diff
@@ -0,0 +1,388 @@
+  var query = datastore.createQuery('Task')
+    .filter('tag =', 'fun')
+    .filter('tag =', 'programming');
+  // [END array_value_equality_range]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testInequalityRange = function(callback) {
+  var datastore = this.datastore;
+
+  // [START inequality_range]
+  var query = datastore.createQuery('Task')
+    .filter('created >', new Date('1990-01-01T00:00:00z'))
+    .filter('created <', new Date('2000-12-31T23:59:59z'));
+  // [END inequality_range]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testInequalityInvalid = function(callback) {
+  var datastore = this.datastore;
+
+  // [START inequality_invalid]
+  var query = datastore.createQuery('Task')
+    .filter('priority >', 3)
+    .filter('created >', new Date('1990-01-01T00:00:00z'));
+  // [END inequality_invalid]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testEqualAndInequalityRange = function(callback) {
+  var datastore = this.datastore;
+
+  // [START equal_and_inequality_range]
+  var query = datastore.createQuery('Task')
+    .filter('priority =', 4)
+    .filter('done =', false)
+    .filter('created >', new Date('1990-01-01T00:00:00z'))
+    .filter('created <', new Date('2000-12-31T23:59:59z'));
+  // [END equal_and_inequality_range]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testInequalitySort = function(callback) {
+  var datastore = this.datastore;
+
+  // [START inequality_sort]
+  var query = datastore.createQuery('Task')
+    .filter('priority >', 3)
+    .order('priority')
+    .order('created');
+  // [END inequality_sort]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testInequalitySortInvalidNotSame = function(callback) {
+  var datastore = this.datastore;
+
+  // [START inequality_sort_invalid_not_same]
+  var query = datastore.createQuery('Task')
+    .filter('priority >', 3)
+    .order('created');
+  // [END inequality_sort_invalid_not_same]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testInequalitySortInvalidNotFirst = function(callback) {
+  var datastore = this.datastore;
+
+  // [START inequality_sort_invalid_not_first]
+  var query = datastore.createQuery('Task')
+    .filter('priority >', 3)
+    .order('created')
+    .order('priority');
+  // [END inequality_sort_invalid_not_first]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testLimit = function(callback) {
+  var datastore = this.datastore;
+
+  // [START limit]
+  var query = datastore.createQuery('Task')
+    .limit(5);
+  // [END limit]
+
+  this.datastore.runQuery(query, callback);
+};
+
+Query.prototype.testCursorPaging = function(callback) {
+  var pageSize = 1;
+
+  datastore.createQuery = this.datastore.createQuery;
+
+  // [START cursor_paging]
+  var query = datastore.createQuery('Task')
+    // Pagination is enabled by default in gcloud-node. To switch it off:
```

Comment:
I think this is slightly different. The `autoPaginate` command seems really more of an "autoBatch" feature. Pagination is particularly for users wanting to show pages of results on their site. In this case, they should set the limit to their pageSize and let gcloud get all the results of that page. However, we need to show them how to use cursors and the more_results information to show a "Next Page" link on their site.